### PR TITLE
 Bob refunds swap after restart that requires communication

### PR DESCRIPTION
**Pull request by D4nte**
As Bob is dialing Alice, we now ensure that we are connected to Alice
at each step that needs communication.
If we are not connected, we proceed with dialing.

In an attempt to improve libp2p usage, we also add known address of
Alice first and only use peer_id to dial.
This ensures that we use the expected peer id.

### CODE REVIEW

**Review by rishflab — APPROVED** [review]
LGTM!

**Comment by D4nte** [comment]
> LGTM!
> 
> This PR means we no longer have to work on this issue #89 or merge this PR #101?

Yes for #101. Not sure about #89.

File: swap/src/cli.rs
Comment (da-kami) [inline swap/src/cli.rs:84]: ❗️ Please remove the option `p2p-address` (line 109) below in favor of these options. As far as I can see this is duplicated now. Note that for convenience it would help to keep  the `default` value, potentially also for the `alice_addr` above.

Please add another comment similar to the one on option `p2p-address` - the connection options are actually only relevant for Bob (there is a mistake in the current comment) because he has to be able to dial Alice. 
Since `Resume` is used by both parties this is a bit confusing but we can deal with it later.

File: swap/tests/happy_path.rs
Comment (da-kami) [inline swap/tests/happy_path.rs:40]: 🙃 Would potentially be nicer to let `init_alice` return the connection information she will use. That way we can change this code in one place, independent of each test.

File: swap/tests/happy_path_restart_bob.rs
Comment (da-kami) [inline swap/tests/happy_path_restart_bob.rs:264]: ❓ Is this code still relevant?

File: swap/tests/happy_path_restart_bob.rs
Comment (da-kami) [inline swap/tests/happy_path_restart_bob.rs:292]: ❗️ Nice! I think we have to do this for the other restart tests as well otherwise they might fail. This relates to what was discussed here: https://github.com/comit-network/xmr-btc-swap/pull/89

**Review by da-kami — CHANGES_REQUESTED** [review]
Nice work! 
I would vote for fixing the CLI `resume` command and changing the other restart tests to wait for the other party properly before we merge.

File: swap/tests/happy_path.rs
Comment (rishflab) [reply swap/tests/happy_path.rs:40]: I think we did this or planned to do it #101

File: swap/src/cli.rs
Comment (rishflab) [inline swap/src/cli.rs:50]: I don't think it makes to have  peer-id in the CLI if we don't have a way to configure multiple multiaddr's for a peer.

**Review by rishflab — CHANGES_REQUESTED** [review]

File: swap/src/cli.rs
Comment (D4nte) [reply swap/src/cli.rs:50]: Multiple multiaddr for a CLI sounds like a feature.

**Comment by da-kami** [comment]
I thought with making Alice's listening port random we would not have to use separate files anymore. Why was https://github.com/comit-network/xmr-btc-swap/pull/100/commits/1e4b1155564edbf479d57d9a28b0e778c292c2c7 needed? 
Is the logic of `get_port` to get a random unused port?

**Comment by D4nte** [comment]
> I thought with making Alice's listening port random we would not have to use separate files anymore. Why was [1e4b115](https://github.com/comit-network/xmr-btc-swap/commit/1e4b1155564edbf479d57d9a28b0e778c292c2c7) needed?
> Is the logic of `get_port` to get a random unused port?

It looked like I had performance issues in the CI with two tests in one file.

**Review by da-kami — APPROVED** [review]
We can do the following things in follow-ups: 

* All tests that restart have to properly wait until all parties have finished execution before asserting balances.
* Cleanup the CLI `resume` command (duplicated address information).
* Investigate why we still need separate files for executing tests, #101 had fixed that by using https://docs.rs/portpicker/0.1.0/portpicker/, but we closed in in favor of this PR using https://docs.rs/get-port/3.0.0/get_port/

**Comment by da-kami** [comment]
> > I thought with making Alice's listening port random we would not have to use separate files anymore. Why was [1e4b115](https://github.com/comit-network/xmr-btc-swap/commit/1e4b1155564edbf479d57d9a28b0e778c292c2c7) needed?
> > Is the logic of `get_port` to get a random unused port?
> 
> It looked like I had performance issues in the CI with two tests in one file.

Interesting. I would have guessed the performance would get better if we had multiple test in one file.

File: swap/src/cli.rs
Comment (D4nte) [reply swap/src/cli.rs:50]: What I mean is that it looks like that can be a future feature.

File: swap/src/cli.rs
Comment (D4nte) [reply swap/src/cli.rs:84]: Done in https://github.com/comit-network/xmr-btc-swap/pull/102

File: swap/tests/happy_path_restart_bob.rs
Comment (D4nte) [reply swap/tests/happy_path_restart_bob.rs]: Done

**Pull request merged**
